kafka/adminapi.go

```diff
@@ -1505,6 +1528,22 @@ func (a *AdminClient) cConfigResourceToResult(cRes **C.rd_kafka_ConfigResource_t
 	return result, nil
 }
 
+// cToDeletedRecordResult converts a C topic partitions list to a Go DeleteRecordsResult slice.
+func cToDeletedRecordResult(cparts *C.rd_kafka_topic_partition_list_t) (results []DeleteRecordsResult) {
+	partcnt := int(cparts.cnt)
+	results = make([]DeleteRecordsResult, partcnt)
+	partitions := newTopicPartitionsFromCparts(cparts)
+
+	for i := 0; i < partcnt; i++ {
```

Comment:
Let's delegate all C handling to `newTopicPartitionsFromCparts`
```suggestion
	partitions := newTopicPartitionsFromCparts(cparts)
	partitionsLen := len(partitions)
	results = make([]DeleteRecordsResult, partitionsLen)

	for i := 0; i < partitionsLen; i++ {
```